gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -1450,16 +1451,14 @@ gst_tensor_src_iio_start (GstBaseSrc * src)
   g_free (device_name);
 
   self->buffer_data_fp = g_new (struct pollfd, 1);
-  self->buffer_data_file = fopen (filename, "r");
-  if (self->buffer_data_file == NULL) {
+  self->buffer_data_fp->fd = open (filename, O_RDONLY | O_NONBLOCK);
+  g_free (filename);
```

Comment:
you are accessing filename later, Line 1458

Reply:
updated